server/backend/api/app/crud/crud_user.py

```diff
@@ -246,9 +247,142 @@ def fetch_initial_requester_info(
+    only for IDIR and Business BCeID users, ignore bc service card users
+    """
+    # get a requester from the database
+    requester = (
+        db.query(models.FamUser)
+        .filter(models.FamUser.user_name == "CMENG")
+        .one_or_none()
+    )
+    # setup IDIM web service
+    api_instance_env = (
+        ApiInstanceEnv.PROD if crud_utils.is_on_aws_prod() else ApiInstanceEnv.TEST
+    )
+    idim_proxy_service = IdimProxyService(requester, api_instance_env)
+
+    # grab fam users from user table
+    fam_users = get_users(db)
+    total_users_count = len(fam_users)
+    LOGGER.debug(f"Total number of users: {total_users_count}")
+    if use_pagination:
+        fam_users = (
+            db.query(models.FamUser)
+            .order_by(models.FamUser.user_id.asc())
+            .offset((page - 1) * per_page)
+            .limit(per_page)
+            .all()
+        )
+        LOGGER.debug(
+            f"Updating information for users on page {page}, there are {per_page} users on each page"
+        )
+
+    success_user_list = []
+    failed_user_list = []
+    ignored_user_list = []
+
+    for user in fam_users:
+        try:
+            LOGGER.debug(
+                f"Updating information for user: {user.user_name}, type: {user.user_type_code}, guid: {user.user_guid}"
+            )
+            search_result = None
+            properties_to_update = {}
+
+            if user.user_type_code == UserType.IDIR:
+                # IDIM web service doesn't support search IDIR by user_guid, so we search by userID
+                search_result = idim_proxy_service.search_idir(
+                    schemas.IdimProxySearchParam(**{"userId": user.user_name})
+                )
+
+                if not user.user_guid:
+                    # if user has no user_guid in our database, add it
+                    properties_to_update = {
+                        models.FamUser.user_guid: search_result.get("guid"),
+                    }
+
+                # if found user's user_guid does not match our record
+                # which is the edge case that could cause by the username change, ignore this situation
+                # our auth lambda and grant access apis will cover this situation
+
+            elif user.user_type_code == UserType.BCEID:
+                if user.user_guid:
+                    # if found business bceid user by user_guid, update username if necessary
+                    search_result = idim_proxy_service.search_business_bceid(
+                        schemas.IdimProxyBceidSearchParam(
+                            **{
+                                "searchUserBy": IdimSearchUserParamType.USER_GUID,
+                                "searchValue": user.user_guid,
+                            }
+                        )
+                    )
+                    properties_to_update = {
+                        models.FamUser.user_name: search_result.get("userId"),
+                    }
```

Comment:
Same as previous comment:

When you later have the if search_result.get("found") when search_result is **found**, just add this "userId" to update there, same as "guid" and all other fields?